### PR TITLE
[Feature] support creating materialized view on jdbc external table

### LINKED ISSUE #19581: Support create materialized view on jdbc external table
## Feature request

In current version(3.0), materialized view could be created on internal table, external table for Hive/Iceberg/Hudi, but not on other kinds of external table like Jdbc. 

So here we propose to support these kinds of external table, to make the materialized view as hub of data federation.

### PR DESCRIPTION
**Pull request by murphyatwork**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #19581

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by LiShuMing** [comment]
Do you know how to rewrite jdbc external table mv? just like external table's mv?

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java [hunk 1]
```diff
@@ -104,6 +107,14 @@
 
 public class MaterializedViewAnalyzer {
 
+    private static final Set<Table.TableType> SUPPORTED_TABLE_TYPE =
+            ImmutableSet.of(Table.TableType.OLAP,
+                    Table.TableType.HIVE,
+                    Table.TableType.HUDI,
+                    Table.TableType.ICEBERG,
+                    Table.TableType.JDBC,
+                    Table.TableType.MYSQL);
```
Comment (satanson) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java:116]: how does SR perceive new data of JDBC/MYSQL table?

Comment (mofeiatwork) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java:116]: we don't guarantee the freshness of jdbc table

**Review by dirtysalt — APPROVED** [review]

**Review by ABingHuang — APPROVED** [review]

**Comment by mofeiatwork** [comment]
@Mergifyio backport 3.0

**Comment by mofeiatwork** [comment]
https://github.com/Mergifyio backport branch-3.0

**Pull request merged**